src/commands/force/user/password/generate.ts

```diff
@@ -45,13 +69,44 @@ export class UserPasswordGenerateCommand extends SfdxCommand {
         const connection: Connection = await Connection.create({ authInfo });
         const org = await Org.create({ connection });
         const user: User = await User.create({ org });
-        const password = User.generatePasswordUtf8();
+        let passwordCondition: PasswordConditions;
+        if (connection.metadata) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const list: any = await connection.metadata.list({ type: 'ProfilePasswordPolicy' });
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-explicit-any
+          const profPasswordPolicies: ProfilePasswordPolicy = JSON.parse(
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any, @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            JSON.stringify(await connection.metadata.read('ProfilePasswordPolicy', [list.fullName]))
+          );
+          passwordCondition = {
+            length: profPasswordPolicies.minimumPasswordLength,
+            complexity: profPasswordPolicies.passwordComplexity,
+          };
+
+          if (!passwordCondition.length || !passwordCondition.complexity) {
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment, @typescript-eslint/no-unsafe-member-access
+            const orgPasswordPolicy: OrgPasswordPolicy = JSON.parse(
+              JSON.stringify(await connection.metadata.read('SecuritySettings', ['passwordPolicies']))
+            );
+            passwordCondition.length = passwordCondition.length
+              ? passwordCondition.length
+              : orgPasswordPolicy.passwordPolicies?.minimumPasswordLength;
+            // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+            passwordCondition.complexity = passwordCondition.complexity
+              ? passwordCondition.length
+              : PasswordComplexityTypes[orgPasswordPolicy.passwordPolicies.complexity];
+          }
+        } else
+          passwordCondition = {
```

Comment:
since the argument is optional and defaulted in sfdxCore, you don't have to do this here.  It can be undefined, right?